triage/experiments/base.py

```diff
@@ -452,8 +462,15 @@ def catwalk(self):
 
     def run(self):
         try:
-            self.build_matrices()
-        finally:
-            logging.info('Matrix build done or errored, cleaning up state table')
-            self.state_table_generator.clean_up()
+            try:
+                logging.info('Building matrices')
+                self.build_matrices()
+            finally:
+                logging.info('Cleaning up state table')
+                with timeout(self.cleanup_timeout):
+                    self.state_table_generator.clean_up()
+        except Exception:
+            logging.exception('Run interrupted by uncaught exception')
+            raise
```

Comment:
To be clear, the _only_ required change is to nest the block to be given a timeout in `with timeout…`, (line 470).

I added the encompassing try/except logging to be sure that the caller is given a useful error message and traceback; but, I'm not 100% clear as to how this method is invoked and what makes sense to users. (Is this log message superfluous? _Etc._) It does make sense in captured test output, (when the test is made to fail), but that's not a typical use, (and the tests don't actually inspect or care about the logs).

Reply:
hmmm... @thcrock might have a better sense than me about whether the additional logging message is superfluous here, though I have had trouble in the past with exceptions not always getting logged from running an experiment, so it might make sense to include here. Though I think we'd want to log the exception itself, which I don't think this will do, will it?

Reply:
`logging.exception` includes the full traceback, including _both_ exceptions, if both `build_matrices` and `clean_up` raise them. That's why it was important to _me_, in development anyway; but, it's possible it's superfluous to the end user.

Reply:
ah, nice -- I thought you had to pass the exception object to it in order for it to be logged. In that case, I think I'd be inclined to keep the logging from the user perspective as well. Though is there any reason no to move the `self.catwalk()` call inside the `try... except` block as well?

Reply:
I have a habit of yelling at folks who log the exception object, since then, though you get the exception object `str`, you lose its traceback 😸 

As for `catwalk` – sure. I hadn't been considering it. It's probably too well-programmed to have errors, though?

Reply:
oh, haha -- in that case I've been using `logging.exception()` to do exactly the opposite of what I wanted!

As for `catwalk`, I think I've had a couple of PRs merged, so we better be sure to have robust exception handling at least for those parts of the code...

Reply:
exception-logging extended as requested 👍